frontend/src/screens/wallet/receive/ReceiveOffer.tsx

```diff
@@ -0,0 +1,140 @@
+import { AlertTriangleIcon, CopyIcon } from "lucide-react";
+import React from "react";
+import { Link } from "react-router-dom";
+import QRCode from "src/components/QRCode";
+import {
+  Alert,
+  AlertDescription,
+  AlertTitle,
+} from "src/components/ui/alert.tsx";
+import { Button } from "src/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardHeader,
+  CardTitle,
+} from "src/components/ui/card";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useToast } from "src/components/ui/use-toast";
+
+import { copyToClipboard } from "src/lib/clipboard";
+import { CreateOfferRequest } from "src/types";
+import { request } from "src/utils/request";
+
+export default function ReceiveOffer() {
+  const { toast } = useToast();
+  const [isLoading, setLoading] = React.useState(false);
+  const [description, setDescription] = React.useState<string>("");
+  const [offer, setOffer] = React.useState<string | null>(null);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    try {
+      setLoading(true);
+      const offer = await request<string>("/api/offers", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          description,
+        } as CreateOfferRequest),
+      });
+
+      if (offer) {
+        setOffer(offer);
+
+        toast({
+          title: "Successfully created BOLT-12 offer",
+        });
+      }
+    } catch (e) {
+      toast({
+        variant: "destructive",
+        title: "Failed to create offer: " + e,
+      });
+      console.error(e);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const copy = () => {
+    copyToClipboard(offer as string, toast);
+  };
+
+  return (
+    <div className="grid gap-5">
+      <Alert>
+        <AlertTriangleIcon className="h-4 w-4" />
+        <AlertTitle>BOLT-12 is in beta phase</AlertTitle>
```

Comment:
Not sure this is the right wording. Maybe the title and description can be reviewed

Reply:
Changed a bit, can you check now?